pkg/scalers/github_runner_scaler.go

```diff
@@ -0,0 +1,646 @@
+	var err error
+	var owner string
+	switch runnerScope {
+	case ORG:
+		owner, err = getValueFromMetaOrEnv("owner", config.TriggerMetadata, config.ResolvedEnv, "ORG_NAME")
+	case ENT:
+		owner, err = getValueFromMetaOrEnv("owner", config.TriggerMetadata, config.ResolvedEnv, "ENTERPRISE_NAME")
+	default:
+		owner, err = getValueFromMetaOrEnv("owner", config.TriggerMetadata, config.ResolvedEnv, "")
+	}
+	if err != nil {
+		return "", err
+	}
+	return owner, nil
+}
+
+// getInt64ValueFromMetaOrEnv returns the value of the given key from the metadata or the environment variables
+func getInt64ValueFromMetaOrEnv(key string, config *ScalerConfig, defVal int64) (int64, error) {
+	sInt, err := getValueFromMetaOrEnv(key, config.TriggerMetadata, config.ResolvedEnv, "")
+	if err != nil {
+		return defVal, nil
+	}
+
+	goodInt, err := strconv.ParseInt(sInt, 10, 64)
+	if err != nil {
+		return -1, fmt.Errorf("error parsing %s: %w", key, err)
+	}
+	return goodInt, nil
+}
+
+func parseGitHubRunnerMetadata(config *ScalerConfig) (*githubRunnerMetadata, error) {
+	meta := githubRunnerMetadata{}
+	meta.targetWorkflowQueueLength = defaultTargetWorkflowQueueLength
+
+	if val, err := getValueFromMetaOrEnv("runnerScope", config.TriggerMetadata, config.ResolvedEnv, "RUNNER_SCOPE"); err == nil && val != "" {
+		meta.runnerScope = val
+	} else {
+		return nil, err
+	}
+
+	if val, err := getOwner(meta.runnerScope, config); err == nil && val != "" {
+		meta.owner = val
+	} else {
+		return nil, err
+	}
+
+	if val, err := getInt64ValueFromMetaOrEnv("targetWorkflowQueueLength", config, defaultTargetWorkflowQueueLength); err == nil && val != -1 {
+		meta.targetWorkflowQueueLength = val
+	} else {
+		return nil, err
+	}
+
+	if val, err := getInt64ValueFromMetaOrEnv("activationTargetWorkflowQueueLength", config, defaultActivationTargetWorkflowQueueLength); err == nil && val != -1 {
+		meta.activationTargetWorkflowQueueLength = val
+	} else {
+		return nil, err
+	}
+
+	if val, err := getValueFromMetaOrEnv("labels", config.TriggerMetadata, config.ResolvedEnv, "LABELS"); err == nil && val != "" {
+		meta.labels = strings.Split(val, ",")
+	}
+
+	if val, err := getValueFromMetaOrEnv("repos", config.TriggerMetadata, config.ResolvedEnv, ""); err == nil && val != "" {
+		meta.repos = strings.Split(val, ",")
+	}
+
+	if val, err := getValueFromMetaOrEnv("githubApiURL", config.TriggerMetadata, config.ResolvedEnv, "GITHUB_API_URL"); err == nil && val != "" {
+		meta.githubAPIURL = val
+	} else {
+		meta.githubAPIURL = defaultGithubAPIURL
+	}
+
+	if val, ok := config.AuthParams["personalAccessToken"]; ok && val != "" {
+		// Found the personalAccessToken in a parameter from TriggerAuthentication
+		meta.personalAccessToken = val
+	} else {
+		if val, err := getValueFromMetaOrEnv("personalAccessToken", config.TriggerMetadata, config.ResolvedEnv, "ACCESS_TOKEN"); err == nil && val != "" {
+			meta.personalAccessToken = val
+		} else {
+			return nil, err
+		}
+	}
```

Comment:
We shouldn't allow to set sensitive data in triggerMetadata.
I also think that as the agent authentication doesn't use the PAT, doesn't make sense to read it from environment.

WDYT if we only allow TriggerAuthentication for providing the PAT?

Reply:
The runner doesn't require the access token to *run*, but it does  require an access token to register itself and create the "runner token" that it uses to run. Otherwise the runner wouldn't be able to scale as the tokens are individual.

This usually means that the runner itself will have an ENV variable called ACCESS_TOKEN for this purpose anyway.

I suggest we follow your guidance and remove the setting of PAT in the manifest and allow ENV or Trigger Only, in case you wish to maintain multiple PATs for different scenarios (separate token scopes for register and REST API).

Reply:
If it's an env var, make sure to follow the naming pattern that we have - THanks!